storm-core/src/jvm/org/apache/storm/windowing/WaterMarkEventGenerator.java

```diff
@@ -126,4 +135,20 @@ private void checkFailures() {
     public void start() {
         this.executorFuture = executorService.scheduleAtFixedRate(this, interval, interval, TimeUnit.MILLISECONDS);
     }
+
+    public void shutdown() {
+        LOG.debug("Shutting down WaterMarkEventGenerator");
+        if (this.executorService != null) {
```

Comment:
Nit: This doesn't seem like it can be null?

Reply:
Same here. Will address.